LoRaEngine/LoraKeysManagerFacade/LoRaDeviceCache.cs

```diff
@@ -14,20 +16,15 @@ public sealed class LoRaDeviceCache : IDisposable
 
         private readonly ILoRaDeviceCacheStore cacheStore;
         private readonly string gatewayId;
-        private readonly string devEUI;
+        private readonly DevEui devEUI;
         private readonly string cacheKey;
 
         public bool IsLockOwner { get; private set; }
 
         private string lockKey;
 
-        public LoRaDeviceCache(ILoRaDeviceCacheStore cacheStore, string devEUI, string gatewayId)
+        public LoRaDeviceCache(ILoRaDeviceCacheStore cacheStore, DevEui devEUI, string gatewayId)
         {
-            if (string.IsNullOrEmpty(devEUI))
```

Comment:
I don't think an "empty" devEUI would be allowed. Is this a case we should validate? Do we have a representation of "empty" (all 0)?

Reply:
Since `DevEui` is a value type we could use `default` for that. In this case it is already validated higher up in the stack (at the beginning of each function that calls this code).

Reply:
According to the [DevEUI documentation on the LoRa Developer Portal][deveui-doc]:

> Note: Values of all ones or all zeros ( 0xFFFFFFFFFFFF or 0x0) are not valid EUI-64 values and should indicate that a value for the EUI-64 is not available (null). Do not use these values for the DevEUI or JoinEUI.
>
> &mdash; <cite>[DevEUI documentation on the LoRa Developer Portal][deveui-doc]</cite>, &copy; Semtech Corp.

So yes, a `default(DevEui)` would be invalid. However, since it's a value type, we should be using `DevEui?` to represent a DevEUI that's missing.

[deveui-doc]: https://lora-developers.semtech.com/documentation/tech-papers-and-guides/the-book/deveui/

Reply:
As discussed, the empty and `0xFF..` cases are already validated on all paths that call this method (where we parse the query/path parameters of the Azure Function).